### ISSUE TITLE
Build failure on Ubuntu 18.04

### ISSUE THREAD

**Issue by MDTech-us-MAN** [Astron/Astron#414]
Hello,

I get the following error when building on Ubuntu 18.04:

```
<>/Astron/src/util/filesystem.h:96:19: error: ignoring return value of ‘char* getcwd(char*, size_t)’, declared with attribute warn_unused_result [-Werror=unused-result]
             getcwd(path, PATH_MAX - 1);
             ~~~~~~^~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
```

According to the [man pages](http://man7.org/linux/man-pages/man2/getcwd.2.html), `getcwd` returns a `char *` that is `NULL` if there is a problem.

-----

Additionally, SOCI doesn't build because of the following error:

```
In file included from /usr/include/soci/soci.h:33:0,
                 from <>/Astron/src/database/SociSQLDatabase.cpp:12:
/usr/include/soci/rowset.h:157:16: error: ‘template<class> class std::auto_ptr’ is deprecated [-Werror=deprecated-declarations]
     const std::auto_ptr<statement> st_;
                ^~~~~~~~
In file included from /usr/include/c++/7/memory:80:0,
                 from <>/Astron/src/util/Datagram.h:9,
                 from <>/Astron/src/util/DatagramIterator.h:2,
                 from <>/Astron/src/database/DBOperation.h:7,
                 from <>/Astron/src/database/DatabaseBackend.h:2,
                 from <>/Astron/src/database/OldDatabaseBackend.h:2,
                 from <>/Astron/src/database/SociSQLDatabase.cpp:1:
/usr/include/c++/7/bits/unique_ptr.h:51:28: note: declared here
   template<typename> class auto_ptr;
                            ^~~~~~~~
In file included from /usr/include/soci/soci.h:33:0,
                 from <>/Astron/src/database/SociSQLDatabase.cpp:12:
/usr/include/soci/rowset.h:158:16: error: ‘template<class> class std::auto_ptr’ is deprecated [-Werror=deprecated-declarations]
     const std::auto_ptr<T> define_;
                ^~~~~~~~
In file included from /usr/include/c++/7/memory:80:0,
                 from <>/Astron/src/util/Datagram.h:9,
                 from <>/Astron/src/util/DatagramIterator.h:2,
                 from <>/Astron/src/database/DBOperation.h:7,
                 from <>/Astron/src/database/DatabaseBackend.h:2,
                 from <>/Astron/src/database/OldDatabaseBackend.h:2,
                 from <>/Astron/src/database/SociSQLDatabase.cpp:1:
/usr/include/c++/7/bits/unique_ptr.h:51:28: note: declared here
   template<typename> class auto_ptr;
                            ^~~~~~~~
In file included from /usr/include/soci/soci.h:34:0,
                 from <>/Astron/src/database/SociSQLDatabase.cpp:12:
/usr/include/soci/session.h:43:41: error: ‘template<class> class std::auto_ptr’ is deprecated [-Werror=deprecated-declarations]
     void set_query_transformation_(std::auto_ptr<details::query_transformation_function> qtf);
                                         ^~~~~~~~
In file included from /usr/include/c++/7/memory:80:0,
                 from <>/Astron/src/util/Datagram.h:9,
                 from <>/Astron/src/util/DatagramIterator.h:2,
                 from <>/Astron/src/database/DBOperation.h:7,
                 from <>/Astron/src/database/DatabaseBackend.h:2,
                 from <>/Astron/src/database/OldDatabaseBackend.h:2,
                 from <>/Astron/src/database/SociSQLDatabase.cpp:1:
/usr/include/c++/7/bits/unique_ptr.h:51:28: note: declared here
   template<typename> class auto_ptr;
                            ^~~~~~~~
In file included from /usr/include/soci/soci.h:34:0,
                 from <>/Astron/src/database/SociSQLDatabase.cpp:12:
/usr/include/soci/session.h: In member function ‘void soci::session::set_query_transformation(T)’:
/usr/include/soci/session.h:80:14: error: ‘template<class> class std::auto_ptr’ is deprecated [-Werror=deprecated-declarations]
         std::auto_ptr<details::query_transformation_function> qtf(new details::query_transformation<T>(callback));
              ^~~~~~~~
In file included from /usr/include/c++/7/memory:80:0,
                 from <>/Astron/src/util/Datagram.h:9,
                 from <>/Astron/src/util/DatagramIterator.h:2,
                 from <>/Astron/src/database/DBOperation.h:7,
                 from <>/Astron/src/database/DatabaseBackend.h:2,
                 from <>/Astron/src/database/OldDatabaseBackend.h:2,
                 from <>/Astron/src/database/SociSQLDatabase.cpp:1:
/usr/include/c++/7/bits/unique_ptr.h:51:28: note: declared here
   template<typename> class auto_ptr;
                            ^~~~~~~~
cc1plus: all warnings being treated as errors
```

**Comment by Psychotropos** [comment]
The first build error should be fixed as of commit 15e1dd3.

The second build error has to do with the SOCI library itself, and is thus not within our control. You can, however, get around it by adding `-Wno-deprecated-declarations` to your build flags.

**Comment by Psychotropos** [comment]
Fair enough. That, too should be fixed as of 1ef2c68. Could you confirm that to be the case?

**Comment by Psychotropos** [comment]
Closing as (presumably) resolved.
